source/CMakeLists.txt

```diff
@@ -13,7 +13,6 @@ IF( NOT BIG_ENDIAN )
     ADD_DEFINITIONS( "-D__LITTLE_ENDIAN__" )
```

Comment:
Maybe use target_compile_definitions with PRIVATE/PUBLIC/INTERFACE depending on what is needed, then these are also forwarded, and the ADD_DEFINITIONS in SIOConfig could be removed.

Reply:
@rete I suppose INTERFACE is the right choice here for the three definitions of `__LITTLE_ENDIAN__`, `SIO_LOGLVL` and `SIO_MACROS_WITH_EXCEPTIONS`? Or do I misunderstand the `INTERFACE` property once again?

Reply:
No you are right. This is where these variables should end up indeed. Can you add them too?

Reply:
Also check with a nice `make VERBOSE=1` that the flags are correctly passed during compilation

Reply:
Done. Also tried briefly using the `SIO::sio` target in a downstream package and the definitions appear in the compiler invocations.

Reply:
I think PUBLIC would be the equivalent what was there before. With INTERFACE, SIO itself wouldn't be compiled with these defines, with PUBLIC SIO and things linking against it, with PRIVATE only SIO itself.

Reply:
Yes, you are right it has to be PUBLIC. Changed.